docs/src/literate/ns_vs_diffeq.jl

```diff
@@ -0,0 +1,463 @@
+
+    cache.invγdt = inv(dt * nlsolver.γ)
+    cache.tstep = integrator.t + nlsolver.c * dt
+    OrdinaryDiffEq.calculate_residuals!(weight, fill!(weight, one(eltype(u))), uprev, u,
+                         opts.abstol, opts.reltol, opts.internalnorm, t);
+
+    # Before starting the nonlinear solve we have to set the time correctly.
+    # Note that ch is a global variable for now.
+    #+
+    update!(ch, cache.tstep);
+
+    # The update of u takes uprev + z or tmp + z most of the time, so we have
+    # to enforce Dirichlet BCs here. Note that these mutations may break the
+    # error estimators.
+    #+
+    apply!(uprev, ch)
+    apply!(tmp, ch)
+    apply_zero!(z, ch);
+
+    nothing
+end;
+
+# For the linear equations we can cleanly integrate with the linear solver
+# interface provided by the DifferentialEquations ecosystem. We use a direct
+# solver for simplicity, altough it comes with some issues. Implementing
+# GMRES with efficient preconditioner is left open for future work.
+mutable struct FerriteLinSolve{CH,F}
+    ch::CH
+    factorization::F
+    A
+end
+FerriteLinSolve(ch) = FerriteLinSolve(ch,lu,nothing)
+function (p::FerriteLinSolve)(::Type{Val{:init}},f,u0_prototype)
+    FerriteLinSolve(ch)
+end
+function (p::FerriteLinSolve)(x,A,b,update_matrix=false;reltol=nothing, kwargs...)
+    if update_matrix
+        ## Apply Dirichlet BCs
+        apply_zero!(A, b, p.ch)
+        ## Update factorization
+        p.A = p.factorization(A)
+    end
+    ldiv!(x, p.A, b)
+    apply_zero!(x, p.ch)
+    return nothing
+end;
+
+# DifferentialEquations assumes dense matrices by default, which is not
+# feasible for semi-discretization of finize element models. We communicate
+# that a sparse matrix with specified pattern should be utilized through the
+# `jac_prototyp` argument. Additionally, we have to provide the mass matrix.
+# To apply the nonlinear portion of the Navier-Stokes problem we simply hand
+# over the dof handler to the right hand side as a parameter in addition to
+# the pre-assembled linear part (which is time independent) to save some
+# runtime.
+jac_sparsity = sparse(K)
+function navierstokes!(du,u,p,t)
+    K,dh,cellvalues_v = p
+    du .= K * u
+
+    n_basefuncs = getnbasefunctions(cellvalues_v)
+
+    ## Nonlinear contribution
+    for cell in CellIterator(dh)
+        Ferrite.reinit!(cellvalues_v, cell)
+        ## Trilinear form evaluation
+        all_celldofs = celldofs(cell)
+        v_celldofs = all_celldofs[dof_range(dh, :v)]
+        v_cell = u[v_celldofs]
+        for q_point in 1:getnquadpoints(cellvalues_v)
+            dΩ = getdetJdV(cellvalues_v, q_point)
+            ∇v = function_gradient(cellvalues_v, q_point, v_cell)
+            v = function_value(cellvalues_v, q_point, v_cell)
+            for j in 1:n_basefuncs
+                φⱼ = shape_value(cellvalues_v, q_point, j)
+                # Note that the order the gradient term is now on the left, which is the correct thing to do here.
```

Comment:
```suggestion
                # Note that the other the gradient term is now on the left, which is the correct thing to do here.
```
not sure, but I guess `other`